sstables/mx/reader.cc

```diff
@@ -429,7 +429,10 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
             if (ret == consumer_m::row_processing_result::skip_row) {
                 _state = state::FLAGS;
                 auto current_pos = position() - _processing_data->size();
-                co_yield skip(*_processing_data, _next_row_offset - current_pos);
+                auto maybe_skip_bytes = skip(*_processing_data, _next_row_offset - current_pos);
+                if (std::holds_alternative<skip_bytes>(maybe_skip_bytes)) {
+                    co_yield maybe_skip_bytes;
```

Comment:
maybe std::move(maybe_skip_bytes)? Or does the compiler optimize this automatically somehow?

Reply:
I don't think it's optimized right now, but ```skip_bytes``` only has 1 int, so I don't think it makes a big difference